core/src/transport/upgrade.rs

```diff
@@ -23,35 +23,33 @@ use multiaddr::Multiaddr;
 use crate::{
     transport::Transport,
     upgrade::{
-        OutboundUpgrade,
-        InboundUpgrade,
+        Upgrade,
         apply_inbound,
         apply_outbound,
         UpgradeError,
-        OutboundUpgradeApply,
-        InboundUpgradeApply
+        InboundUpgradeApply,
+        OutboundUpgradeApply
     }
 };
 use tokio_io::{AsyncRead, AsyncWrite};
 
 #[derive(Debug, Copy, Clone)]
-pub struct Upgrade<T, U> { inner: T, upgrade: U }
+pub struct UpgradeTransport<T, U> { inner: T, upgrade: U }
```

Comment:
I like this change, the old name was confusing. I'd propose going one step further and call it `TransportUpgrade` (and rename the file `transport_upgrade.rs`); the phrase "Create an UpgradeTransport and pass it along." sounds wrong, whereas "Create a TransportUpgrade and pass it along." works fine.